### PR TITLE
Document MQTT availability_mode

### PR DESCRIPTION
**Pull request by emontnemery**
## Proposed change

Document MQTT availability_mode.

Remaining components documented in #16210

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/44987

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

**Comment by emontnemery** [comment]
@Koenkk Can you help review please, I think it became a bit hard to read: https://deploy-preview-16158--home-assistant-docs.netlify.app/integrations/alarm_control_panel.mqtt/#availability_mode

File: source/_integrations/alarm_control_panel.mqtt.markdown [hunk 1]
```diff
@@ -56,6 +56,11 @@ availability:
       description: An MQTT topic subscribed to receive availability (online/offline) updates.
       required: true
       type: string
+availability_mode:
+  description: When `availability` is configured, this controls the conditions needed to set the entity to `available`. Valid entries are `all`, `any`, and `latest`. If set to `all` a valid `payload_available` must be received on all configured availability topics before the entity is marked as online. If set to `any` a valid `payload_available` must be received on at least one configured availability topics before the entity is marked as online. If set to `latest`, the last valid `payload_available` or `payload_not_available` received on any configured availability topic controls the availability.
+  required: false
+  type: string
+  default: latest
```
Comment (Koenkk) [inline source/_integrations/alarm_control_panel.mqtt.markdown:63]: In my opinion this describes the logic well.

Some textual suggestions:
- If set to `all` a valid `payload_available`        -> If set to `all` a `payload_available`
- If set to `any` a valid `payload_available` -> If set to `any` a `payload_available`
- one configured availability topics -> one configured availability topic      (without the trailing s)
- the last valid payload_available -> the last valid payload_available
- the last valid payload_available -> the last payload_available

Comment (emontnemery) [reply source/_integrations/alarm_control_panel.mqtt.markdown:63]: Thanks! Updated.

**Review by frenck — APPROVED** [review]
Thanks! 👍

**Pull request merged**